eZ/Publish/API/Repository/Tests/SetupFactory/Legacy.php

```diff
@@ -177,11 +179,17 @@ public function insertData()
         $data['ezmedia'] = array();
         $data['ezkeyword'] = array();
 
-        foreach ($data as $table => $rows) {
-            // Cleanup before inserting (using TRUNCATE for speed, however not possible to rollback)
-            $q = $dbPlatform->getTruncateTableSql($handler->quoteIdentifier($table));
-            $connection->executeUpdate($q);
+        foreach (array_reverse(array_keys($data)) as $table) {
+            try {
+                // Cleanup before inserting (using TRUNCATE for speed, however not possible to rollback)
+                $connection->executeUpdate($dbPlatform->getTruncateTableSql($handler->quoteIdentifier($table)));
+            } catch (DBALException | PDOException $e) {
+                // Fallback to DELETE if TRUNCATE failed (because of FKs for instance)
```

Comment:
Assuming here that this exception is already handled in code, and that's why it's only added in the test, not in code.